lib/biker.rb

```diff
@@ -1,3 +1,38 @@
 class Biker
+  attr_reader :name, 
+              :max_distance, 
+              :rides, 
+              :acceptable_terrain
 
+  def initialize(name, max_distance)
+    @name = name
+    @max_distance = max_distance
+    @rides = Hash.new()
+    @acceptable_terrain = []
+  end
+
+  def learn_terrain!(terrain)
+    @acceptable_terrain << terrain
+  end
+
+  def log_ride(ride_object, time)
+    if @acceptable_terrain.include?(ride_object.terrain) && ride_object.total_distance <= @max_distance
```

Comment:
One way around nested if statements here is to use a guard clause at the beginning (`return nil if . . .`)